basic.go

```diff
@@ -69,3 +90,81 @@ func (m *mapper) mapUint(numVal reflect.Value, inlineable bool) (nodeID, string)
 	m.nodeSummaries[getNodeKey(numVal)] = "uint"
 	return m.newBasicNode(numVal, printed), "uint"
 }
+
+func (m *mapper) getNodeID(iVal reflect.Value) nodeID {
+	// have to key on kind and address because a struct and its first element have the same UnsafeAddr()
+	key := getNodeKey(iVal)
+	var id nodeID
+	var ok bool
+	if id, ok = m.nodeIDs[key]; !ok {
+		id = nodeID(len(m.nodeIDs))
+		m.nodeIDs[key] = id
+		return id
+	}
+
+	return id
+}
+
+func (m *mapper) newBasicNode(iVal reflect.Value, text string) nodeID {
+	id := m.getNodeID(iVal)
+	fmt.Fprintf(m.writer, "  %d [label=\"<name> %s\"];\n", id, text)
+	return id
+}
+
+func (m *mapper) mapValue(iVal reflect.Value, parentID nodeID, inlineable bool, depth uint32) (nodeID, string) {
+	if !iVal.IsValid() {
+		// zero value => probably result of nil pointer
+		return m.nodeIDs[nilKey], m.nodeSummaries[nilKey]
+	}
+
+	key := getNodeKey(iVal)
+	if summary, ok := m.nodeSummaries[key]; ok {
+		// already seen this address so no need to map again
+		return m.nodeIDs[key], summary
+	}
+	switch iVal.Kind() {
+	// Indirections
+	case reflect.Ptr, reflect.Interface:
+		return m.mapPtrIface(iVal, inlineable, depth)
+	// Collections
+	case reflect.Struct:
+		if m.config.maxDepth > 0 && depth > m.config.maxDepth {
```

Comment:
I'm pretty sure we have interpreted `depth` differently. 😆 When you say "number of graphviz nodes", do you mean a) in total? or b) the number of graphviz nodes down the embedded tree of the root object?


> i.e. if this value is going to cause another node to be created and we've already reached the limit then stop. This is perhaps a bit tricky because of the behaviour of inlineable though.

I think this actually caught my attention during the first testing, and I worked around it. If it's inlineable it will still be rendered, `maxdepth` "only" has an influence on next level objects. Please go in and test this, I will too, I am unclear on the details 😉